lib/Discovery_Schema.cpp

```
@@ -2241,3 +2335,676 @@ int Discovery_Schema::log_query_tool_call(
+	result->add_column_definition(SQLITE_TEXT, "sum_time");
+	result->add_column_definition(SQLITE_TEXT, "min_time");
+	result->add_column_definition(SQLITE_TEXT, "max_time");
+
+	pthread_rwlock_rdlock(&mcp_digest_rwlock);
+
+	for (auto const& [key1, inner_map] : mcp_digest_umap) {
+		for (auto const& [digest, stats_ptr] : inner_map) {
+			MCP_Query_Digest_Stats* stats = (MCP_Query_Digest_Stats*)stats_ptr;
+			char** pta = (char**)malloc(sizeof(char*) * 10);
+
+			pta[0] = strdup(stats->tool_name.c_str());                         // tool_name
```

Cppcheck (2.19.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)